### ISSUE TITLE
Azurite doesn't support snapshot parameter in the Get Block List operation on block blob

### ISSUE THREAD

**Issue by zzhxiaofeng** [Azure/Azurite#636]
**Error Description:**
In Azure, it will return the error "InvalidQueryParameterValue" when use Get Block List operation with invalid snapshot.
But in Azurite, it can get block list successfully in this case.
Azure error is shown as following:
![Screenshot 2020-11-23 172529](https://user-images.githubusercontent.com/20970631/100210493-026f5900-2f46-11eb-9014-cefdeaeabc73.png)

**To Reproduce**
Please run the demo code as following:
```
import {
  BlobServiceClient
} from "@azure/storage-blob";
import * as assert from "assert"

async function main() {
  const blobServiceClient = await BlobServiceClient.fromConnectionString("<Azurite-https-connectionString>");
  const containerClient = await blobServiceClient.getContainerClient("<container-name>");
  const blockBlobClient = await containerClient.getBlockBlobClient("<blob-name>");
  const version = "invalidsnapshot";
  await blockBlobClient.createSnapshot({metadata:{version:version}});
  const snapshotBlobClient = blockBlobClient.withSnapshot(version);
  try {
    await snapshotBlobClient.getBlockList("all");
  } catch (err) {
    console.log(err);
    assert.ok((err as any).response.parsedBody.code === "InvalidQueryParameterValue");
  }
}

main();
```

**Error Track:**
The reason maybe is that Get Block List handler method doesn't use passed snapshot in Azurite. The code is shown as following:
[https://github.com/Azure/Azurite/blob/master/src/blob/handlers/BlockBlobHandler.ts#L352](https://github.com/Azure/Azurite/blob/master/src/blob/handlers/BlockBlobHandler.ts#L352)
![Screenshot 2020-11-18 173832](https://user-images.githubusercontent.com/20970631/99512884-f70bb300-29c4-11eb-8f3c-f159e224822a.png)

**Expected Behavior**
When use Get Block List operation with invalid snapshot, it can return error "InvalidQueryParameterValue".

@jongio for notification.

**Comment by blueww** [comment]
@zzhxiaofeng
Thanks for raising the issue!
We will evaluate it.

We welcome contribution to Azurite.
It would be great if you could contribute the the fix of this issue.

**Comment by blueww** [comment]
This is fixed on Azurite v3.10.0.

**Comment by v-xuto** [comment]
azsdk-azurite
